packages/headless/src/features/folding/folding-state.ts

```diff
@@ -0,0 +1,32 @@
+import {Result} from '../../api/search/search/result';
+
+export interface FoldedResult extends Result {
+  /**
+   * The unsorted children of this result.
+   */
+  children: FoldedResult[];
```

Comment:
I improvised this doc, I wanted it to convey that those results aren't necessarily sorted by date.

Reply:
Would it be more helpful to say `The children of the result, sorted by relevance`?

Also, why do we have two interfaces extending `Result` in the PR?

Reply:
> Would it be more helpful to say `The children of the result, sorted by relevance`?

It might be, I was hesitant on whether to mention it or not, partly since I'm not sure what this relevance is relative to.

> why do we have two interfaces extending `Result` in the PR?

One of them is to represent fields returned by the API, the other is to represent a field generated by the folding reducer.

Reply:
From the design doc, my understanding was the sort is based on the query (e.g. keywords in search box, facet selections). Let's make sure ofc before documenting this.

If the child order is truly random though, then I think we should leave out the word `unsorted`, and just say `The children of the result.`

Reply:
I just tested and it seems like those are sorted in the same order as the search results. I'll update the documentation to reflect that.